frontend/src/app/service/http-service/transformation-http.service.ts

```diff
@@ -40,6 +41,57 @@ export class TransformationHttpService {
     return this.httpClient.put<Transformation>(url, transformation);
   }
 
+  public updateExpandComponent(
+    transformation: Transformation
+  ): Observable<Transformation> {
+    const url = `${this.apiEndpoint}/transformations/${transformation.id}?update_component_code=true&expand_component_code=true`;
+    return this.httpClient.put<Transformation>(url, transformation);
+  }
+
+  public unitTestComponent(
+    transformation: Transformation
+  ): Observable<UnitTestResults> {
+    const url = `${this.apiEndpoint}/transformations/${transformation.id}/test`;
+    return this.httpClient.post<UnitTestResults>(url, {});
+  }
+
+  public importTrafoRevFromString(
+    trafo_revs_str: string,
+    updateCode: boolean,
+    expandCode: boolean,
+    overwriteReleased: boolean
+  ): Observable<Response> {
+    let import_obj: TrafoStringMixed[];
+
+    try {
+      const parsed_json = JSON.parse(trafo_revs_str) as Transformation;
+
+      if (Array.isArray(parsed_json)) {
+        import_obj = parsed_json;
+      } else {
+        import_obj = [parsed_json];
+      }
+    } catch (error) {
+      console.warn('Failed to parse Trafo(s) JSON:', error);
+      import_obj = [trafo_revs_str];
+    }
+
+    let params = new HttpParams();
+
+    if (updateCode) {
+      params = params.append('update_component_code', 'true');
+    }
+    if (expandCode) {
+      params = params.append('expand_component_code', 'true');
+    }
+    if (overwriteReleased) {
+      params = params.append('allow_overwrite_released', 'true');
+    }
+
+    const url = `${this.apiEndpoint}/transformations/?update_component_code=true&expand_component_code=true`;
```

Comment:
Diese QueryParams gehören oben ins HttpParams-Objekt und nicht in den URL-String.